modules/serverless-kn-func-invoke-reference.adoc

```diff
@@ -0,0 +1,111 @@
+* `--data`: Content for the request. For CloudEvent requests, this is the CloudEvent `data` attribute.
+* `--file`: Path to a local file containing data to be sent.
+* `--content-type`: The MIME content type for the request.
+* `-p`, `--path`: Path to the project directory.
+* `-c`, `--confirm`: Prompt to interactively confirm all options.
+* `-v`, `--verbose`: Print verbose output.
+* `-h`, `--help`: Prints information of usage of `kn func invoke`.
+
+In particular, you might find it useful to specify the following parameters:
+
+Event target:: The target instance of the invoked function. Accepts the `local` value for a locally deployed function, the `remote` value for a remotely deployed function, or a URL for a function deployed at an arbitrary endpoint. If target is not specified, it defaults to `local`.
+Event message format:: The message format, such as `http` or `cloudevent`. Defaults to the format of the template that was used when creating the function.
+Event type:: The type of event being sent. You can find information about the `type` parameter that is set for events from a certain event producer in the documentation for that event producer. For example, the API server source may set the `type` parameter of produced events as `dev.knative.apiserver.resource.update`.
+Event source:: The unique event source that produced the event. This may be a URI for the event source, for example `https://10.96.0.1/`, or the name of the event source.
+Event ID:: A random, unique ID that is created by the event producer.
+Event data:: Allows you to specify a `data` value for the event sent by the `kn func invoke` command. For example, you can specify a `--data` value such as `"Hello World"` so that the event contains this data string. By default, no data is included in the events created by `kn func invoke`.
++
+[NOTE]
+====
+Functions that have been deployed to a cluster can respond to events from an existing event source that provides values for properties such as `source` and `type`. These events often have a `data` value in JSON form, which captures the domain specific context of the event. Using the CLI flags noted in this document, developers can simulate those events for local testing.
+====
++
+You can also send event data using the `--file` flag to provide a local file containing data for the event. Make sure to specify the content type using `--content-type`.
+Data content type:: If you are using the `--data` flag to add data for events, you can also specify what type of data is carried by the event, by using the `--content-type` flag. In the previous example, the data is plain text, so you might specify `kn func invoke --data "Hello world!" --content-type "text/plain"`.
+
+.Example commands specifying event parameters by using flags
+
+[source,terminal]
+----
+$ kn func invoke --type <event_type> --source <event_source> --data <event_data> --content-type <content_type> --id <event_ID> --format <format> --namespace <namespace>
+----
+
```

Comment:
You could consider adding a description before all code blocks, describing what each example does explicitly. This will ensure that the reader has a clear understanding about each example. The same applies where multiple code blocks are not separated by descriptions in the PR, but I won't mention them all in this review.